crates/manifest/src/schema/v2.rs

```diff
@@ -145,11 +174,116 @@ pub struct Component {
     /// Settings for custom tools or plugins. Spin ignores this field.
     #[serde(default, skip_serializing_if = "Map::is_empty")]
     pub tool: Map<String, toml::Table>,
+    /// If true, allow dependencies to inherit configuration.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub dependencies_inherit_configuration: Option<bool>,
+    /// Component dependencies
+    #[serde(default, skip_serializing_if = "ComponentDependencies::is_empty")]
+    pub dependencies: ComponentDependencies,
+}
+
+/// Component dependencies
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(try_from = "Map<DependencyName, ComponentDependency>")]
+pub struct ComponentDependencies {
+    /// `dependencies = { "foo:bar" = ">= 0.1.0" }`
+    pub inner: Map<DependencyName, ComponentDependency>,
+}
+
+impl TryFrom<Map<DependencyName, ComponentDependency>> for ComponentDependencies {
+    type Error = anyhow::Error;
+
+    fn try_from(value: Map<DependencyName, ComponentDependency>) -> Result<Self, Self::Error> {
+        let dependencies = ComponentDependencies { inner: value };
+        dependencies.validate()?;
+        Ok(dependencies)
+    }
+}
+
+impl ComponentDependencies {
+    fn validate(&self) -> anyhow::Result<()> {
+        self.ensure_package_names_no_export()?;
+        self.ensure_disjoint()?;
+        Ok(())
+    }
+
+    /// This method ensures that dependency names in the package form (e.g. "foo:bar" or "foo:bar@0.1.0") do not have an export specified.
```

Comment:
```suggestion
    /// Whole-package imports (e.g. "foo:bar" or "foo:bar@0.1.0") cannot map to specific exported interfaces.
```

(I struggled with the current wording because it seemed like "do not have an export specified" was referring to the dependency name and I was going "well duh."  Not sure my suggestion is quite right but hopefully inspires something better!)

Reply:
Updated the wording to make it more explicit that the right-hand-side of a `DependencyName` cannot have an export specified. LMKWYT